File: Books/The-Sixth-Edition-Of-Programmer-Interview-Canon/16.11.py
"""
面试题 16.11. 跳水板
简单
https://leetcode-cn.com/problems/diving-board-lcci/
"""

from typing import List

class Solution:
    # 方法一: 先算所有都是最短的，然后依次减去一个短的，加上一个长的。但是注意k为0和（短木板和长木板相等的情况）

    # 80 ms	    62.37%
    # 17.7 MB   100.00%
    def divingBoard(self, shorter: int, longer: int, k: int) -> List[int]:
        if k == 0:
            return []
        if shorter == longer:
            return [shorter * k]
        res = []
        minLength = shorter * k
        res.append(minLength)
        last = minLength
        for i in range(k):
            last -= shorter
            last += longer
            res.append(last)
        return res

File: Leetcode-Swift/程序员面试金典(第6版)/16.11.swift
//
//  16.11.swift
//  Leetcode-Swift
//
//  Created by devcz on 2021/12/4.
//

import Foundation

/*
 面试题 16.11. 跳水板 (简单)
 https://leetcode-cn.com/problems/diving-board-lcci/
 */

class Solution_jindian_1611 {
    func divingBoard(_ shorter: Int, _ longer: Int, _ k: Int) -> [Int] {
        // 校验k的值
        if k <= 0 {
            return []
        }
        
        // 校验shorter和longer
        if shorter == longer {
            return [shorter * k]
        }
        
        // 总弄有k+1种结果
        var ans = [Int](repeating: 0, count: k + 1)
        // 第1个为全都是短木板
        ans[0] = shorter * k
        
        // 依次减去短木板并增加长木板
        var last = ans[0]
        for i in 1..<(k+1) {
            last -= shorter
            last += longer
            ans[i] = last
        }
        
        return ans
    }
    
    func test() {
        print(divingBoard(1, 1, 0))
        print(divingBoard(3, 3, 2))
        print(divingBoard(3, 4, 3))
    }
}
